### PR TITLE
Don't test inplace_merge with libc++

### PR DESCRIPTION
**Pull request by msimberg**
AFAICT [this](https://git.codingcafe.org/Mirrors/llvm-mirror/libcxx/commit/25a78dcd773ce509469a2b892adbeac0c230cdb9) is not yet in any released version of libc++ so I'm ignoring the test for all versions of libc++.

I'll update this to check the version as well once the fix has been merged.

"Fixes" #2964.

### CODE REVIEW

File: tests/unit/parallel/algorithms/inplace_merge.cpp [hunk 1]
```diff
@@ -47,9 +47,11 @@ int hpx_main(boost::program_options::variables_map& vm)
     std::cout << "using seed: " << seed << std::endl;
     std::srand(seed);
 
+#if !defined(_LIBCPP_VERSION)
```
Comment (sithhell) [inline tests/unit/parallel/algorithms/inplace_merge.cpp]: IIUC, it's only a problem up until version 5.

**Comment by msimberg** [comment]
Updated to allow the test with libc++ version >= 6 as it runs successully on the pycicle clang 6 builder.

**Comment by msimberg** [comment]
Hmm, I think disabling the test completely is almost as bad as having it always pass.

I've now disabled only the stability tests, as the others should still pass. But I also added a cmake test for it (`HPX_HAVE_STABLE_INPLACE_MERGE`). @hkaiser @taeguk What do you think about this?

File: cmake/tests/stable_inplace_merge.cpp [hunk 2]
```diff
@@ -0,0 +1,15 @@
+////////////////////////////////////////////////////////////////////////////////
+//  Copyright (c) 2017 Mikael Simberg
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+////////////////////////////////////////////////////////////////////////////////
+
+#if defined(_LIBCPP_VERSION) && (_LIBCPP_VERSION < 6)
+#  error "libc++ inplace_merge implementation is broken"
+#endif
```
Comment (sithhell) [inline cmake/tests/stable_inplace_merge.cpp]: Instead of checking the stdlib version, couldn't we actually test if inplace_merge is actually stable?

Comment (msimberg) [reply cmake/tests/stable_inplace_merge.cpp]: Right, that does sound like the correct thing to do. Need to make sure it's thorough enough to catch future bugs in `inplace_merge` as well though, so to be honest I think the current check is simple and good enough. (libc++ is unlikely to have this bug again if they've added good tests for it, but then we need not trust them of course...)

**Comment by taeguk** [comment]
@msimberg LGTM, too!

Apart from this PR, the user can still use ```inplace_merge```.
Is there need to restrict or warn for using ```inplace_merge``` with libc++ < 6 ?

**Pull request merged**